packages/kbn-config-schema/src/types/string_type.ts

```diff
@@ -29,18 +29,33 @@ export type StringOptions = TypeOptions<string> & {
 
 export class StringType extends Type<string> {
   constructor(options: StringOptions = {}) {
-    let schema = internals.string().allow('');
+    // We want to allow empty strings, however calling `allow('')` casues
+    // Joi to whitelist the value and skip any additional validation.
+    // Instead, we reimplement the string validator manually except in the
+    // hostname case where empty strings aren't allowed anyways.
+    let schema =
+      options.hostname === true
+        ? internals.string().hostname()
+        : internals.any().custom(value => {
```

Comment:
Will we have to implement similar logic for all advanced string validation rules, like `ip`, `domain`, `email`, etc?

Reply:
Unfortunately :-/